.github/workflows/docker-build-test.yaml

```diff
@@ -307,17 +301,116 @@ jobs:
+                            sed 's/.*refs\/heads\///' | 
+                            sort -t. -k2,2n -k3,3n | 
+                            tail -n 1)
+          elif [[ $base_branch =~ ^aptos-release-v([0-9]+)\.([0-9]+) ]]; then
+            major=${BASH_REMATCH[1]}
+            minor=${BASH_REMATCH[2]}
+
+            if [[ $minor -eq 0 ]]; then
+                # If the minor version is 0, look for the latest branch in the previous major version
+                prev_major=$((major - 1))
+                target_branch=$(git ls-remote --heads origin "aptos-release-v${prev_major}.*" | 
+                                grep -E "refs/heads/aptos-release-v${prev_major}\.[0-9]+$" | 
+                                sed 's/.*refs\/heads\///' | 
+                                sort -t. -k2,2n -k3,3n | 
+                                tail -n 1)
+            else
+                # If the minor version is greater than 0, look for the previous minor version
+                prev_minor=$((minor - 1))
+                target_branch="aptos-release-v${major}.${prev_minor}"
+            fi
+
+            # Now do the check whether the target branch exists, outside the if-else
+            if git ls-remote --heads origin "aptos-release-v${target_branch}" | grep -q "refs/heads/${target_branch}"; then
+                echo "Target branch is $target_branch"
+            else
+                echo "Branch $target_branch does not exist"
+            fi
+          else
+            echo "Not merging to main or a release branch, using the main branch for compat test" >&2
+            target_branch="main"
+          fi
+          echo "TARGET_BRANCH=${target_branch}" >> $GITHUB_OUTPUT
+          echo "Determined TARGET_BRANCH: ${target_branch}" >&2
+
+      # actions/get-latest-docker-image-tag requires docker utilities and having authenticated to internal docker image registries
+      - uses: aptos-labs/aptos-core/.github/actions/docker-setup@main
+        id: docker-setup
+        with:
+          GCP_WORKLOAD_IDENTITY_PROVIDER: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+          GCP_SERVICE_ACCOUNT_EMAIL: ${{ secrets.GCP_SERVICE_ACCOUNT_EMAIL }}
+          EXPORT_GCP_PROJECT_VARIABLES: "false"
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DOCKER_ARTIFACT_REPO: ${{ secrets.AWS_DOCKER_ARTIFACT_REPO }}
+          GIT_CREDENTIALS: ${{ secrets.GIT_CREDENTIALS }}
+
+      - name: Get Docker Image Tag
+        uses: ./.github/actions/get-latest-docker-image-tag
+        id: get-docker-image-tag
+        with:
+          branch: ${{ steps.determine-the-target-branch.outputs.TARGET_BRANCH }}
+          variants: "failpoints performance"
+
+      - name: Log TARGET_BRANCH and IMAGE_TAG
+        run: |
+          echo "TARGET_BRANCH: ${{ steps.determine-the-target-branch.outputs.TARGET_BRANCH }}"
```

Comment:
A nice way to display this as well is adding it to the github step summary, then it will show up on the summary page as a note while the job is running

```
echo TARGET ... >> $GITHUB_STEP_SUMMARY
```

This can also be done from typescript / anywhere in your actions / forge etc